internal/config/config.go

```diff
@@ -397,6 +404,54 @@ func resolveCollector(allowedDirs []string) (*Collector, error) {
 	return col, nil
 }
 
+// generate self-signed certificate for OTEL receiver
+// nolint: revive
+func handleSelfSignedCertificates(col *Collector) error {
+	if col.Receivers.OtlpReceivers != nil {
+		for _, receiver := range col.Receivers.OtlpReceivers {
+			if receiver.OtlpTLSConfig != nil && receiver.OtlpTLSConfig.GenerateSelfSignedCert {
+				err := processOtlpReceivers(receiver.OtlpTLSConfig)
+				if err != nil {
+					return fmt.Errorf("failed to generate self-signed certificate: %w", err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func processOtlpReceivers(tlsConfig *OtlpTLSConfig) error {
+	sanNames := []string{"127.0.0.1", "::1", "localhost"}
```

Comment:
should these be config default values like DefCollectorTLSCAPath, DefCollectorTLSCertPath and DefCollectorTLSKeyPath?

Reply:
Go won't let me use a slice for a const.  OK if do something like
`DefCollectorTLSSANNames          = "127.0.0.1, ::1, localhost"` 
and then
`sanNames := strings.Split(DefCollectorTLSSANNames, ",")` ?